src/components/views/read/cvForm/Template.tsx

```diff
@@ -1,130 +1,195 @@
+    const { files } = e.target;
+    const file = files?.[0];
+
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = () => {
+        const dataUrl = reader.result;
+        setUrl(dataUrl as string);
+        setUploadedFile(file);
+      };
+      reader.readAsDataURL(file);
+    } else {
+      // If no file is selected, reset the uploaded file state
+      setUploadedFile(null);
+    }
+  };
+
+  const handleFileRemove = (): void => {
+    setVisibility(false);
+    setUrl('');
+    setUploadedFile(null);
+    if (inputRef.current) {
+      inputRef.current.value = '';
+    }
+  };
+
+  const handleVisibility = (): void => {
+    setVisibility(!visibility);
+  };
   const handlePrev = (): void => {
     prevPage();
     prevStep();
   };
-  const [templates, setTemplates] = useState([
-    { id: 1, thumbnailink: 'hello', title: 'professional template' },
-    { id: 2, thumbnailink: '2nd', title: 'normal template' },
-    { id: 3, thumbnailink: '3rd', title: 'modern template' },
-  ]);
-  const [templateStates, setTemplateStates] = useState<TemplateState[]>(
-    templates.map((template) => ({ id: template.id, showFields: false })),
-  );
-  const handleDone = (templateId: number): void => {
-    const updatedTemplateStates = templateStates.map((templateState) => {
-      if (templateState.id === templateId) {
-        return { ...templateState, showFields: false };
-      }
-      return templateState;
-    });
-    setTemplateStates(updatedTemplateStates);
-  };
-  const handleNext = (): void => {
-    // const modifiedValues = {...values, 'Template': {selectedTemplate}}
-    // handleValues(modifiedValues);
-    nextPage();
-    nextStep();
-  };
+
+  const personalData = cvValues.personalInfo;
+  const educationData = cvValues.educationInfo;
+  const workData = cvValues.workInfo;
+  const skillsData = cvValues.skillsInfo;
+  const portfolioData = cvValues.portfolioInfo;
+  const motivationData = cvValues.motivationInfo;
+  const referencesData = cvValues.referencesInfo;
   return (
-    <div>
-      <h2>Template</h2>
-      {templates.map((template, index) => (
-        <Card
-          key={template.id}
-          sx={{
-            maxWidth: 900,
-            position: 'relative',
-            zIndex: 100,
-            height: '400px',
-            overflow: 'auto',
-          }}
-          className={`card-item card-${template.id}`}
-        >
+    <Box>
+      {templates.map((template) => (
+        <Card key={template.id}>
           <CardContent>
             <Typography gutterBottom variant="h5" component="div">
               {template.title} - {template.id}
             </Typography>
             <Typography variant="body2" color="text.secondary">
               Select A Template
             </Typography>
-            {templateStates[index].showFields && (
-              <img alt={template.title} src={template.thumbnailink} />
+            {template.title === 'professional template' && (
```

Comment:
Don't do this. Define your `templates` as an array of objects. Each object hols the info of one template. One key of that template object can be a react component. Then use that component to render the CV.

Reply:
I will work on fixing that, but I guess it will be also better to discuss it next week.